File: perl/eg/light.pl
use Time::HiRes;
use lib "..";
use ev3;

print "Waiting the EV3 brick online...\n";
ev3::ev3_init() > 0 || exit( 1 );

print "*** ( EV3 ) Hello! ***\n";

ev3::set_light_blink( $ev3::LIT_RIGHT, $ev3::LIT_GREEN, 1000, 500 );
Time::HiRes::sleep( 0.5 );
ev3::set_light_blink( $ev3::LIT_RIGHT, $ev3::LIT_RED, 1000, 500 );

ev3::set_light_blink( $ev3::LIT_LEFT, $ev3::LIT_GREEN, 1000, 1000 );
Time::HiRes::sleep( 1.5 );
ev3::set_light_blink( $ev3::LIT_LEFT, $ev3::LIT_RED, 1000, 1000 );
Time::HiRes::sleep( 7.5 );

ev3::set_light( $ev3::LIT_LEFT, $ev3::LIT_OFF );
ev3::set_light( $ev3::LIT_RIGHT, $ev3::LIT_OFF );
my $col = ev3::light_color( ev3::get_light( $ev3::LIT_LEFT ));
print "LEFT : $col\n";
$col = ev3::light_color( ev3::get_light( $ev3::LIT_RIGHT ));
print "RIGHT: $col\n";
Time::HiRes::sleep( 0.5 );
ev3::set_light( $ev3::LIT_LEFT, $ev3::LIT_RED );
ev3::set_light( $ev3::LIT_RIGHT, $ev3::LIT_RED );
$col = ev3::light_color( ev3::get_light( $ev3::LIT_LEFT ));
print "LEFT : $col\n";
$col = ev3::light_color( ev3::get_light( $ev3::LIT_RIGHT ));
print "RIGHT: $col\n";
Time::HiRes::sleep( 0.5 );
ev3::set_light( $ev3::LIT_LEFT, $ev3::LIT_AMBER );
ev3::set_light( $ev3::LIT_RIGHT, $ev3::LIT_AMBER );
$col = ev3::light_color( ev3::get_light( $ev3::LIT_LEFT ));
print "LEFT : $col\n";
$col = ev3::light_color( ev3::get_light( $ev3::LIT_RIGHT ));
print "RIGHT: $col\n";
Time::HiRes::sleep( 0.5 );
ev3::set_light( $ev3::LIT_LEFT, $ev3::LIT_GREEN );
ev3::set_light( $ev3::LIT_RIGHT, $ev3::LIT_GREEN );
$col = ev3::light_color( ev3::get_light( $ev3::LIT_LEFT ));
print "LEFT : $col\n";
$col = ev3::light_color( ev3::get_light( $ev3::LIT_RIGHT ));
print "RIGHT: $col\n";

ev3::ev3_uninit();
print "*** ( EV3 ) Bye! ***\n";

File: ruby/eg/light.rb
# -*- coding: utf-8 -*-

#FIXME: not the best way to import the library
$LOAD_PATH << if ENV[ 'OS' ] == 'Windows_NT' then '../mingw' else '..' end
require 'ev3'
include Ev3

puts 'Waiting the EV3 brick online...'
if ev3_init() < 1 then exit( 1 ) end

puts '*** ( EV3 ) Hello! ***'

set_light_blink( LIT_RIGHT, LIT_GREEN, 1000, 500 )
sleep( 0.5 )
set_light_blink( LIT_RIGHT, LIT_RED, 1000, 500 )

set_light_blink( LIT_LEFT, LIT_GREEN, 1000, 1000 )
sleep( 1.5 )
set_light_blink( LIT_LEFT, LIT_RED, 1000, 1000 )
sleep( 7.5 )

set_light( LIT_LEFT, LIT_OFF )
set_light( LIT_RIGHT, LIT_OFF )
col = light_color( get_light( LIT_LEFT ))
puts "LEFT : #{col}"
col = light_color( get_light( LIT_RIGHT ))
puts "RIGHT: #{col}"
sleep( 0.5 )
set_light( LIT_LEFT, LIT_RED )
set_light( LIT_RIGHT, LIT_RED )
col = light_color( get_light( LIT_LEFT ))
puts "LEFT : #{col}"
col = light_color( get_light( LIT_RIGHT ))
puts "RIGHT: #{col}"
sleep( 0.5 )
set_light( LIT_LEFT, LIT_AMBER )
set_light( LIT_RIGHT, LIT_AMBER )
col = light_color( get_light( LIT_LEFT ))
puts "LEFT : #{col}"
col = light_color( get_light( LIT_RIGHT ))
puts "RIGHT: #{col}"
sleep( 0.5 )
set_light( LIT_LEFT, LIT_GREEN )
set_light( LIT_RIGHT, LIT_GREEN )
col = light_color( get_light( LIT_LEFT ))
puts "LEFT : #{col}"
col = light_color( get_light( LIT_RIGHT ))
puts "RIGHT: #{col}"

ev3_uninit()
puts '*** ( EV3 ) Bye! ***'
